### ISSUE TITLE
Imbalanced Partition Assignment to Consumers Per Topic

### ISSUE THREAD

**Issue by mehranmeidani** [twmb/franz-go#795]
Hi,

I have three topics with different numbers of partitions, specifically:

- T1 and T2 with 24 partitions each
- T3 with 96 partitions

There are two consumers within a consumer group using the default balancer, both interested in all three topics. However, at any given moment, one consumer has 72 partitions of T3, and the other has the remaining partitions. I understand that each consumer has 72 partitions in total, which seems balanced, but this doesn't work for me because each topic requires different processing. I need the consumers to have an equal number of partitions per topic.

In summary, this is what happens:
- Consumer 1: 72 partitions of T3
- Consumer 2: 24 partitions of T1, 24 partitions of T2, 24 partitions of T3

What I expect:
- Consumer 1: 48 partitions of T3, 12 partitions of T2, 12 partitions of T1
- Consumer 2: 48 partitions of T3, 12 partitions of T2, 12 partitions of T1

Am I making a mistake somewhere?
Thanks.

**Comment by twmb** [comment]
It sounds like you need the roundrobin assignor. The sticky assignor's mandate is to preserve _where_ partitions _were_ being consumed, as much as possible. In this case, 72 partitions were preserved on consumer 1, which is more ideal than in what you'd prefer where only 48 partitions are preserved.

The roundrobin assignor may lead to more shuffling of partitions when new members come and go, but it'll result in equal balancing of topics to consumers.
